core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -706,6 +714,7 @@ object ShuffleBlockFetcherIterator {
    */
   private[storage] case class SuccessFetchResult(
       blockId: BlockId,
+      mapId: Int,
```

Comment:
why we need the map index here？

Reply:
Yeah, that's right. Here we need to guarantee all paths to `throwFetchFailedException` has mapIndex pass though, even a SuccessFetchResult still can trigger fetch failed exception.